kotlinpoet/src/main/java/com/squareup/kotlinpoet/KmSpecs.kt

```diff
@@ -0,0 +1,405 @@
+import com.squareup.kotlinpoet.km.PropertyAccessorFlag.IS_EXTERNAL
+import com.squareup.kotlinpoet.km.PropertyAccessorFlag.IS_INLINE
+import com.squareup.kotlinpoet.km.PropertyAccessorFlag.IS_NOT_DEFAULT
+import com.squareup.kotlinpoet.km.declaresDefaultValue
+import com.squareup.kotlinpoet.km.hasGetter
+import com.squareup.kotlinpoet.km.hasSetter
+import com.squareup.kotlinpoet.km.isAbstract
+import com.squareup.kotlinpoet.km.isAnnotation
+import com.squareup.kotlinpoet.km.isCompanionObject
+import com.squareup.kotlinpoet.km.isConst
+import com.squareup.kotlinpoet.km.isCrossInline
+import com.squareup.kotlinpoet.km.isData
+import com.squareup.kotlinpoet.km.isDeclaration
+import com.squareup.kotlinpoet.km.isDelegated
+import com.squareup.kotlinpoet.km.isDelegation
+import com.squareup.kotlinpoet.km.isEnum
+import com.squareup.kotlinpoet.km.isEnumEntry
+import com.squareup.kotlinpoet.km.isExpect
+import com.squareup.kotlinpoet.km.isExternal
+import com.squareup.kotlinpoet.km.isFakeOverride
+import com.squareup.kotlinpoet.km.isFinal
+import com.squareup.kotlinpoet.km.isInfix
+import com.squareup.kotlinpoet.km.isInline
+import com.squareup.kotlinpoet.km.isInner
+import com.squareup.kotlinpoet.km.isInterface
+import com.squareup.kotlinpoet.km.isInternal
+import com.squareup.kotlinpoet.km.isLateinit
+import com.squareup.kotlinpoet.km.isNoInline
+import com.squareup.kotlinpoet.km.isObject
+import com.squareup.kotlinpoet.km.isOpen
+import com.squareup.kotlinpoet.km.isOperator
+import com.squareup.kotlinpoet.km.isOverride
+import com.squareup.kotlinpoet.km.isOverrideProperty
+import com.squareup.kotlinpoet.km.isPrimary
+import com.squareup.kotlinpoet.km.isPrivate
+import com.squareup.kotlinpoet.km.isProtected
+import com.squareup.kotlinpoet.km.isPublic
+import com.squareup.kotlinpoet.km.isSealed
+import com.squareup.kotlinpoet.km.isSuspend
+import com.squareup.kotlinpoet.km.isSynthesized
+import com.squareup.kotlinpoet.km.isTailRec
+import com.squareup.kotlinpoet.km.isVal
+import com.squareup.kotlinpoet.km.isVar
+import com.squareup.kotlinpoet.km.propertyAccessorFlags
+import com.squareup.kotlinpoet.km.toImmutableKmClass
+import kotlinx.metadata.Flags
+import javax.lang.model.element.Element
+import javax.lang.model.element.ElementKind
+import javax.lang.model.element.PackageElement
+import javax.lang.model.element.TypeElement
+import kotlin.reflect.KClass
+
+/** @return a [TypeSpec] ABI representation of this [KClass]. */
+@KotlinPoetKm
+fun KClass<*>.toTypeSpec(): TypeSpec = java.toTypeSpec()
+/** @return a [TypeSpec] ABI representation of this [KClass]. */
+@KotlinPoetKm
+fun Class<*>.toTypeSpec(): TypeSpec = toImmutableKmClass().toTypeSpec()
+/** @return a [TypeSpec] ABI representation of this [TypeElement]. */
+@KotlinPoetKm
+fun TypeElement.toTypeSpec(): TypeSpec = toImmutableKmClass().toTypeSpec()
+/** @return a [FileSpec] ABI representation of this [KClass]. */
+@KotlinPoetKm
+fun KClass<*>.toFileSpec(): FileSpec = java.toFileSpec()
```

Comment:
Do we care about cases when there are multiple types declared in a file?

Reply:
for right now - no. Would definitely be something to iterate on